nemo_skills/mcp/adapters.py

```diff
@@ -116,3 +135,116 @@ def format(self, tool_call, result):
             "tool_call_id": tool_call["id"],
             "content": json.dumps(result) if not isinstance(result, str) else result,
         }
+
+
+# ==============================
+# CONVERSATION MANAGERS
+# ==============================
+
+
+class CompletionConversationManager(ConversationManager):
+    """Manages conversation history for chat completion models."""
+
+    def add_assistant_response(self, conversation: List[Dict[str, Any]], response: Dict[str, Any]) -> None:
+        """Add an assistant response to the conversation."""
+        message = {"role": "assistant", "content": response["generation"]}
+
+        # Add reasoning content if available
+        if "reasoning_content" in response:
+            message["reasoning_content"] = response["reasoning_content"]
+
+        conversation.append(message)
+
+    def add_tool_results(
+        self,
+        conversation: List[Dict[str, Any]],
+        tool_calls_message: Dict[str, Any],
+        formatted_results: List[Dict[str, Any]],
+    ) -> None:
+        """Add tool call results to the conversation."""
+        # Update the last assistant message with tool calls (completion models need this)
+        if conversation and conversation[-1]["role"] == "assistant":
+            conversation[-1].update(tool_calls_message)
+
+        # Add the formatted tool result messages directly - they're already in the right format
+        conversation.extend(formatted_results)
+
+
+class ResponsesConversationManager(ConversationManager):
+    """Manages conversation history for responses API models."""
+
+    def add_assistant_response(self, conversation: List[Dict[str, Any]], response: Dict[str, Any]) -> None:
+        """Add an assistant response to the conversation using serialized output."""
+        # Use the serialized output from the responses API
+        if "serialized_output" in response:
+            conversation.extend(response["serialized_output"])
+        else:
+            # Fallback to basic message format
```

Comment:
is this fallback how the api is supposed to work? Or does it mean something unexpected is happening and we should maybe throw an error?

Reply:
yeah probably better to throw an error instead